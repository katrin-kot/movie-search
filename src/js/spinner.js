export function hideSpinner() {
  document.querySelector('.spinner').classList.remove('visible');
  document.querySelector('.search-clear').classList.remove('hidden');
}

export function showSpinner() {
  document.querySelector('.search-clear').classList.add('hidden');
  document.querySelector('.spinner').classList.add('visible');
}

export function renderSpinner() {
  const swiper1 = document.querySelector('.search-clear');
  swiper1.insertAdjacentHTML(
    'afterbegin',
    `<div class = 'spinner'><svg class="loader" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 340 340"><circle cx="170" cy="170" r="160" stroke="#E2007C"/>
  <circle cx="170" cy="170" r="135" stroke="#404041"/>
  <circle cx="170" cy="170" r="110" stroke="#E2007C"/>
  <circle cx="170" cy="170" r="85" stroke="#404041"/></svg></div>`,
  );
}
