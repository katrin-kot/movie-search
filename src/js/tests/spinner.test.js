import { renderSpinner, hideSpinner } from '../spinner';

describe('renderSpinner', () => {
  it('Should add spinner to swiper-container', () => {
    document.body.innerHTML = `
    <div class="search-clear"></div>
  `;
    renderSpinner();
    expect(document.body.innerHTML).toMatchSnapshot();
  });
});

describe('hideSpinner', () => {
  it('Shold hide spinner with class visible', () => {
    document.body.innerHTML = `
    <div class = "search-clear hidden"></div>
        <div class = "spinner visible"></div>
        `;
    hideSpinner();
    expect(document.querySelector('.visible')).toBeFalsy();
    expect(document.querySelector('.hidden')).toBeFalsy();
  });
});
