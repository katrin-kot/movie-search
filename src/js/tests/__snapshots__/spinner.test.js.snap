// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renderSpinner Should add spinner to swiper-container 1`] = `
"
    <div class=\\"search-clear\\"><div class=\\"spinner\\"><svg class=\\"loader\\" xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 340 340\\"><circle cx=\\"170\\" cy=\\"170\\" r=\\"160\\" stroke=\\"#E2007C\\"></circle>
  <circle cx=\\"170\\" cy=\\"170\\" r=\\"135\\" stroke=\\"#404041\\"></circle>
  <circle cx=\\"170\\" cy=\\"170\\" r=\\"110\\" stroke=\\"#E2007C\\"></circle>
  <circle cx=\\"170\\" cy=\\"170\\" r=\\"85\\" stroke=\\"#404041\\"></circle></svg></div></div>
  "
`;
